partcad/tests/unit/test_git_clone_retries.py

```
@@ -0,0 +1,127 @@
+
+        ctx = pc.Context()
+        factory = pc.ProjectFactoryGit(ctx, None, test_config_import_git)
+        factory._clone_or_update_repo(repo_url)
+
+    # The number of calls must be initial attempt plus 'max' retries
+    assert mock_clone_from.call_count == test_git_retry_config["max"] + 1
+
+
+def test_project_import_git_clone_retry_then_success():
+    fail_count = 3
+    def side_effect(*args, **kwargs):
```

Ruff (0.8.2)

Missing return type annotation for private function `side_effect`

(ANN202)